src/string.rs

```diff
@@ -10,13 +11,26 @@ use core::{
 
 use crate::Vec;
 
-/// A fixed capacity [`String`](https://doc.rust-lang.org/std/string/struct.String.html)
+/// A possible error value when converting a [`String`] from a UTF-16 byte slice.
+///
+/// This type is the error type for the [`from_utf16`] method on [`String`].
+///
+/// [`from_utf16`]: String::from_utf16
+#[derive(Debug, PartialEq, Eq)]
```

Comment:
```suggestion
#[derive(Debug, Clone, PartialEq, Eq)]
```

Matches the `DecodeUtf16Error` struct.

Reply:
https://doc.rust-lang.org/std/string/struct.FromUtf16Error.html doesn't have `Clone` though.

Reply:
It also doesn't have `PartialEq` and `Eq` for that matter.

Reply:
I was thinking it makes sense to match the inner type of [`DecodeUtf16Error`](https://doc.rust-lang.org/core/char/struct.DecodeUtf16Error.html) for this, since it is separate from `std::string::FromUtf16Error` by necessity of having the `Capacity` variant.

Reply:
Given that `std` doesn't have these I don't think these traits are necessary for this type. I think it makes more sense to follow `std` here given that I already had to remove the `push_utf16` method here which was the original incentive for this PR.

Reply:
Fair enough, sounds good to me!